crypto/engine/eng_dyn.c

```diff
@@ -424,6 +424,7 @@ static int dynamic_load(ENGINE *e, dynamic_data_ctx *ctx)
         ctx->bind_engine = NULL;
         DSO_free(ctx->dynamic_dso);
         ctx->dynamic_dso = NULL;
+        ERR_clear_error();
```

Comment:
I'm a little confused by this line being added here. I thought the DSO_bind_func was just returning a pointer to the dynamic engines bind function.
I would have thought if you were seeing the error you would need to fix it by adding the ERR_clear_error() within the if statement starting at line 482 which is where the actual bind is called.